### PR TITLE
Feat : Added playlistPage + refactoring

### PR DESCRIPTION
**Pull request by RahulSingh9131**
This branch contains
1. creating playlist feature
2. adding videos to playlist
3. styles for corresponding pages
4. context for playlist
5. reducer func. for playlist
6. & some basic refactoring

### CODE REVIEW

File: src/css/screens/videoinlist.css [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+.videoinlist-text h4{
+    margin-bottom: 5px;
+    font-size: var(--normal-font-size);
+}
+.videoinlist-text p{
+    color: var(--text-color-light);
+    font-size: var(--small-font-size);
+}
+.videoinlist-avatar{
+    height: 30px !important;
```
Comment (rohit8686) [inline src/css/screens/videoinlist.css:30]: try avoid using important

File: src/css/screens/videoinlist.css [hunk 2]
```diff
@@ -0,0 +1,46 @@
+.videoinlist{
+    margin-bottom: 40px;
+    width: 270px;
+    position: relative;
+}
+.videoinlist-thumbnail{
+    height: 140px;
```
Comment (rohit8686) [inline src/css/screens/videoinlist.css:7]: Suggestion : try using relative units

**Review by rohit8686 — APPROVED** [review]
- Clean and modular code
- Looks good to merge

**Review by shuntzhere — APPROVED** [review]
Code looks good, couldn't find any mistake. The structure seems fine as well.

**Pull request merged**